core/client/src/main/java/alluxio/client/block/StreamFactory.java

```diff
@@ -43,6 +44,10 @@ private StreamFactory() {} // prevent instantiation
    */
   public static BlockOutStream createLocalBlockOutStream(FileSystemContext context, long blockId,
       long blockSize, WorkerNetAddress address, OutStreamOptions options) {
+    if (NettyClient.isDomainSocketSupported(address)) {
```

Comment:
I would prefer to keep the Configuration based methods outside of `NettyClient` and have it be a factory for creating client bootstraps.